pkg/agent/workspace.go

```diff
@@ -303,6 +304,22 @@ func CloneRepositoryForWorkspace(
 
 	log.Done("Successfully cloned repository")
 
+	// Get .devpodignore files to exclude
+	f, err := os.Open(filepath.Join(workspaceDir, ".devpodignore"))
+	if err != nil {
+		return nil
+	}
+	excludes, err := dockerignore.ReadAll(f)
+	if err != nil {
+		log.Warn(".devpodignore file is invalid : ", err)
+		return nil
+	}
+	// Remove files from workspace content folder
+	for _, exclude := range excludes {
+		os.RemoveAll(filepath.Join(workspaceDir, exclude))
```

Comment:
I'm not sure if this is expected. We're using a bind mount for the devcontainer with local docker and this would delete the excludes on the users machine

Reply:
True, but only from the content folder, i.e. `~/.devpod/context/...`, which I think is fine? since other devpod functionality could use this so deleting would help "ignore" the files elsewhere

Reply:
> True, but only from the content folder, i.e. ~/.devpod/context/...

Yep, we still need to make an exception for local docker and local folder because there it's a direct bind mount to the origin folder

Reply:
ah yes your right, my PR description was wrong I had not tested local dir and local docker. Only local dir with kubernetes, which uses the stream RPCs. The deletion here only happens with a git source (since it's in `CloneRepositoryForWorkspace`). I'm not sure how we can support .devpodignore for the local docker and local dir use case. Unless I symlinked or moved the original folder elsewhere, there will be no way around the bind mount mounting ignored files. But maybe this is fine? Since if you are using local docker and dir, you proboably want everything mounted? What do you think? Can you see a way of supporting?

Reply:
Oh nice, I've totally missed this! Thanks for clarifying